src/v/cluster/rm_stm.cc

```diff
@@ -2020,6 +2020,11 @@ void rm_stm::apply_data(model::batch_identity bid, model::offset last_offset) {
             _log_state.ongoing_set.insert(base_offset);
             _mem_state.estimated.erase(bid.pid);
         }
+
+        if (!_log_state.inflight.contains(bid.pid)) {
+            _log_state.inflight[bid.pid] = 0;
+        }
+        _log_state.inflight[bid.pid]++;
```

Comment:
nit: i really like having clear semantics like this, but this is also three separate index look-ups when you probably need only one.